edge-hub/src/Microsoft.Azure.Devices.Routing.Core/checkpointers/MasterCheckpointer.cs

```diff
@@ -305,24 +305,38 @@ enum EventIds
                 Close,
             }
 
+            // Wrapped logging in 'IsEnabled' because profiling showed that 'GetContextString' call was quite expensive
+            // for 'CommitFinished'. Wrapped the rest only for sake of symmetry
             public static void CreateStart(string id)
             {
-                Log.LogDebug((int)EventIds.CreateStart, "[MasterCheckpointerCreateStart] MasterCheckpointerId: {0}", id);
+                if (Log.IsEnabled(LogLevel.Debug))
```

Comment:
is it better to wrap this logic in a logger;? which means we need to create a new logger class wrapping existing serilog logger in Microsoft.Azure.Devices.Edge.Util.Logger.cs; with a function delegate argument in LogDebug and other Log methods.  More work, but fix it for all usages.  What do you think?

Reply:
I think it is needed here only because one parameter passed to the Log is expensive to compute. In fact I would narrow it down to just that case.. this is quite cumbersome to use otherwise.

Reply:
Yes, I was thinking about a delegate based solution to avoid evaluating the log parameters until it is needed, my concerns were that A) it is more work for a single case (only the CommitFinished is extreme slow, the rest is modified only to have the same structure, maybe that was a mistake by me to do that) and B) in 99% passing log parameters is not expensive, and a delegate based lazy evaluation may give the false impression that I do performance optimization using it, but actually creating a delegate (maybe capturing some variables using closures) would be more expensive than passing (and evaluating) the parameters the old way. 
So I thought that it is cheaper and safer to use this ugly double check (double, because LogDebug will also check inside, but that is too late for me) in this exceptional case. If it becomes more regular that creating logged values are expensive, I agree that the right solution is that you are proposing.